### DISCUSSION TITLE
Security considerations

### DISCUSSION THREAD

**Discussion by creesch** [beehaw-org/ui-project#26] [Ideas]
Didn't make an issue yet as it is more of a general thought at the moment. Lemmy-ui has shown to have had some issues with security, specifically XSS related attack vectors.  I think it is a good idea to consider this sort of thing from the start. 

From how lemmy works I think that we should consider any data from the API that was originally provided by users as "dirty". So it is probably good to sanitize it before it is being used. For example, put it through something like [domPurify](https://github.com/cure53/DOMPurify) at key moments. Any thoughts on this?

**Comment by lionirdeadman** [comment]
I think it's certainly a good idea to sanitize on our end but I think it's worth noting that the BE is also working on more sanitizations (see LemmyNet/lemmy/pull/3708 - link intentional broken to not point people here). Ultimately, the more barriers are up, the better even if neither are perfect.

**Reply by creesch** [reply]
Exactly. Although we should be aware of some drawbacks as well of overdoing it. There is a change of double escaping certain things if things get sanitized in my experience.  I am also not set on using domPurify. It is just the library I am most familiar with as Mozilla often requires it for browser extensions in their store where outside input is involved. 

What I think is important that we do it at the point where we introduce the data in the system. That way it is less of an issue if there are implementation mistakes somewhere down the line that normally could be cause for concern.

**Reply by lionirdeadman** [reply]
Hm, actually, because of the way federation content works, I think we shouldn't do it at the input but instead when showing things. The API should do input sanitation instead.

**Reply by creesch** [reply]
To be clear what I meant by "introduced in the system" is when we fetch data from the API. Where the "system"  is our UI and the underlying code base.  So we are in agreement there I think.
